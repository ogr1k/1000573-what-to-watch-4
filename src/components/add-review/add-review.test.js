import React from "react";
import renderer from "react-test-renderer";
import AddReview from "./add-review.jsx";
import {BrowserRouter} from "react-router-dom";

const film = {
  description: `A former Prohibition-era Jewish gangster returns to the Lower East Side of Manhattan over thirty years later, where he once again must confront the ghosts and regrets of his old life.`,
  director: `Sergio Leone`,
  genre: `Crime`,
  id: 1,
  name: `Once Upon a Time in America`,
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Once_Upon_a_Time_in_America.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/Once_Upon_a_Time_in_America.jpg`,
  rating: 9.9,
  year: 1984,
  ratings: 276395,
  starring: [`Robert De Niro`, `James Woods`, `Elizabeth McGovern`],
  video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  runTime: 500
};


it(`Render Add Review`, () => {

  const tree = renderer
    .create(
        <BrowserRouter>
          <AddReview
            film={film}
            onClick={() => {}}
            onChange={() => {}}
            onSubmit={() => {}}
            error={{}}
            isValid={true}
            isFetching={false}
          />
        </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
