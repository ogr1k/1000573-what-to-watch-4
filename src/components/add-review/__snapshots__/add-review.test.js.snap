// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Add Review 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="add-review__notice add-review__error"
    >
      
    </div>
    <div
      className="movie-card__bg"
    >
      <img
        alt="Once Upon a Time in America"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/1"
              onClick={[Function]}
            >
              Once Upon a Time in America
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height={63}
            src="/img/avatar.jpg"
            width={63}
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="Once Upon a Time in America"
        height={327}
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Once_Upon_a_Time_in_America.jpg"
        width={218}
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[Function]}
    >
      
      <div
        className="rating"
      >
        <div
          className="rating__stars"
          onClick={[Function]}
        >
          <input
            className="rating__input"
            defaultChecked={true}
            defaultValue={0}
            disabled={false}
            id="star-0"
            name="rating"
            type="radio"
          />
          <label
            className="rating__label visually-hidden"
            htmlFor="star-0"
          >
            Rating $
            0
          </label>
          <input
            className="rating__input"
            defaultChecked={false}
            defaultValue={1}
            disabled={false}
            id="star-1"
            name="rating"
            type="radio"
          />
          <label
            className="rating__label "
            htmlFor="star-1"
          >
            Rating $
            1
          </label>
          <input
            className="rating__input"
            defaultChecked={false}
            defaultValue={2}
            disabled={false}
            id="star-2"
            name="rating"
            type="radio"
          />
          <label
            className="rating__label "
            htmlFor="star-2"
          >
            Rating $
            2
          </label>
          <input
            className="rating__input"
            defaultChecked={false}
            defaultValue={3}
            disabled={false}
            id="star-3"
            name="rating"
            type="radio"
          />
          <label
            className="rating__label "
            htmlFor="star-3"
          >
            Rating $
            3
          </label>
          <input
            className="rating__input"
            defaultChecked={false}
            defaultValue={4}
            disabled={false}
            id="star-4"
            name="rating"
            type="radio"
          />
          <label
            className="rating__label "
            htmlFor="star-4"
          >
            Rating $
            4
          </label>
          <input
            className="rating__input"
            defaultChecked={false}
            defaultValue={5}
            disabled={false}
            id="star-5"
            name="rating"
            type="radio"
          />
          <label
            className="rating__label "
            htmlFor="star-5"
          >
            Rating $
            5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
      >
        <textarea
          className="add-review__textarea"
          defaultValue=""
          disabled={false}
          id="review-text"
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            disabled={false}
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;
